### PR TITLE
feat(lambda-nodejs): Allow setting mainFields for esbuild

### PR DESCRIPTION
**Pull request by revmischa**
I’m trying to use the new format=esm bundling with NodejsFunction

I noticed that it bundles the “CJS” version of dependencies not the ESM version (if available). esbuild has some wack defaults - https://esbuild.github.io/api/#main-fields

This causes problems because the CJS versions usually have stuff like require() and they don’t work for tree-shaking.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-nodejs/lib/types.ts [hunk 1]
```diff
@@ -270,6 +270,14 @@ export interface BundlingOptions {
...
    */
   readonly format?: OutputFormat;
+
+  /**
+   * How to determine the entry point for modules.
+   * Try "module,main" to default to ES module versions.
+   *
+   * @default - main,module
+   */
+  readonly mainFields?: string;
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/types.ts:280]: Let's make this a `string[]`.

```suggestion
  /**
   * How to determine the entry point for modules.
   * Try ['module', 'main'] to default to ES module versions.
   *
   * @default ['main', 'module']
   */
  readonly mainFields?: string[];
```

File: packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts [hunk 2]
```diff
@@ -211,6 +211,7 @@ export class Bundling implements cdk.BundlingOptions {
       ...this.props.banner ? [`--banner:js=${JSON.stringify(this.props.banner)}`] : [],
       ...this.props.footer ? [`--footer:js=${JSON.stringify(this.props.footer)}`] : [],
       ...this.props.charset ? [`--charset=${this.props.charset}`] : [],
+      ...this.props.mainFields ? [`--main-fields=${this.props.mainFields}`] : [],
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts:214]:
```suggestion
      ...this.props.mainFields ? [`--main-fields=${this.props.mainFields.join(',')}`] : [],
```

**Review by jogold — COMMENTED** [review]
Thanks @revmischa.

Can you add a test there https://github.com/aws/aws-cdk/blob/7c1795ecd078b2a49ac7b0460762b07d6cfe491b/packages/%40aws-cdk/aws-lambda-nodejs/test/bundling.test.ts#L182

And doc in https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-nodejs/README.md#configuring-esbuild

Please also rename your PR to `feat(lambda-nodejs): <your feat>`

File: packages/@aws-cdk/aws-lambda-nodejs/README.md [hunk 3]
```diff
@@ -254,6 +254,18 @@ new lambda.NodejsFunction(this, 'my-handler', {
 
 Note: A [`tsconfig.json` file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) is required 
 
+## Choosing which version of a dependency to bundle
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/README.md:257]: Let's simply add the option in the section [Configuring `esbuild`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-nodejs/README.md#configuring-esbuild)

something like `mainFields: ['module', 'main'], // <short description and defaults to>`

**Review by jogold — APPROVED** [review]
Thanks @revmischa!

@corymhall LGTM.

**Comment by jogold** [comment]
@revmischa can you merge from master? I think it will fix the build.

**Review by corymhall — APPROVED** [review]

**Pull request merged**